languages/ruby/lib/oso/polar/ffi.rb

```diff
@@ -28,39 +36,34 @@ def self.release(ptr)
           Rust.free(ptr) unless ptr.null?
         end
       end
-      # Wrapper class for QueryEvent FFI pointer + operations.
-      class QueryEvent < ::FFI::AutoPointer
-        def self.release(ptr)
-          Rust.free(ptr) unless ptr.null?
-        end
-      end
       # Wrapper class for Error FFI pointer + operations.
       class Error < ::FFI::AutoPointer
         def self.release(ptr)
           Rust.free(ptr) unless ptr.null?
         end
       end
-      # Wrapper class for Message FFI pointer + operations.
-      class Message < ::FFI::AutoPointer
+      # Wrapper class for Rust strings FFI pointer + operations.
+      class RustString < ::FFI::AutoPointer
         def self.release(ptr)
           Rust.free(ptr) unless ptr.null?
         end
       end
 
-      # Wrapper class for Source FFI pointer.
-      class Source < ::FFI::AutoPointer
-        def self.release(ptr)
-          Rust.free(ptr) unless ptr.null?
-        end
-      end
+      # Defines the result type version of
+      # each of these structs
+      # result(T) => { result: T, error: string }
+      #
+      # We have a bunch more here than in the other language
+      # because
```

Comment:
because?

Reply:
Oh lol I think halfway through the thought I realized we could just delete the other classes :)